rpc/src/calls/personal.rs

```diff
@@ -0,0 +1,266 @@
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * ------------------------------------------------------------------------------
+ */
+
+use accounts::{get_key_dir, Account};
+use client::BlockKey;
+use client::ValidatorClient;
+use jsonrpc_core::{Error, Params, Value};
+use messages::seth::{
+    CreateExternalAccountTxn, EvmPermissions, SethTransaction as SethTransactionPb,
+    SethTransaction_TransactionType,
+};
+use openssl::ec::{EcGroup, EcKey};
+use openssl::nid::Nid;
+use openssl::symm::Cipher;
+use requests::RequestHandler;
+use sawtooth_sdk::messaging::stream::MessageSender;
+use std::fs;
+use std::str::from_utf8;
+use transactions::SethTransaction;
+use transform;
+
+pub fn get_method_list<T>() -> Vec<(String, RequestHandler<T>)>
+where
+    T: MessageSender,
+{
+    vec![
+        ("personal_listAccounts".into(), list_accounts),
+        ("personal_newAccount".into(), new_account),
+        ("personal_unlockAccount".into(), unlock_account),
+        ("personal_importRawKey".into(), import_raw_key),
+    ]
+}
+
+macro_rules! except {
+    ($expr:expr, $msg:expr) => {{
+        $expr.map_err(|err| {
+            error!("{}", err);
+            Error {
+                code: (-32069i64).into(),
+                message: $msg.into(),
+                data: None,
+            }
+        })?
+    }};
+}
+
+#[allow(needless_pass_by_value)]
+pub fn list_accounts<T>(_: Params, client: ValidatorClient<T>) -> Result<Value, Error>
+where
+    T: MessageSender,
+{
+    info!("personal_listAccounts");
+
+    let accounts = client.loaded_accounts();
+    let locked_accounts = accounts.read().unwrap();
+
+    Ok(Value::Array(
+        locked_accounts
+            .iter()
+            .map(|a| transform::hex_prefix(a.address()))
+            .collect::<Vec<_>>(),
+    ))
+}
+
+#[allow(needless_pass_by_value)]
+pub fn new_account<T>(params: Params, client: ValidatorClient<T>) -> Result<Value, Error>
+where
+    T: MessageSender,
+{
+    info!("personal_newAccount");
+
+    let args: Vec<Option<String>> = match params.parse::<Vec<Option<String>>>() {
+        Ok(t) => t,
+        Err(_) => {
+            return Err(Error::invalid_params(
+                "Takes [password: String, moderator: String, permissions: String]",
+            ));
+        }
+    };
+
+    // We get Option<Option<String>> here due the possibility of a client
+    // sending in an incomplete list of optional strings. Unnest into plain
+    // options to fix
+    let password = args.get(0).unwrap_or(&None);
+    let moderator = args.get(1).unwrap_or(&None);
+
+    // TODO: Convert to .transpose() call once it's stabilized
```

Comment:
Don't commit TODOs

Reply:
Are there guidelines on this somewhere? I looked at the contributing guide, and TODOs aren't mentioned. Might be worth adding in?

And I can understand why we wouldn't want TODOs that are just there because someone hasn't gotten around to fixing them yet, but in this case we're blocked from using `.transpose()` until it lands in stable. Do you still think the comment shouldn't be added? If so, would it help if I just removed the `TODO` bit of the comment?

Reply:
It is a convention because in-line comments are almost always the wrong place to track work that needs to be done. In this case, we could create a JIRA ticket to resolve it, but IMO I don't think it's something to even worry about. Trying to track ways that code could be written more succinctly if a particular language feature is adopted seems like a nightmare, unless there is a significant performance improvement expected.